crates/sui-types/src/messages.rs

```diff
@@ -945,6 +978,27 @@ impl Transaction {
             Base64::from_bytes(self.data().tx_signature.as_ref()),
         )
     }
+
+    pub fn new_genesis_transaction(objects: Vec<GenesisObject>) -> Self {
+        GenesisTransaction { objects }
+            .pipe(SingleTransactionKind::Genesis)
+            .pipe(TransactionKind::Single)
+            .pipe(|kind| {
+                TransactionData::new(
+                    kind,
+                    SuiAddress::default(),
+                    (ObjectID::ZERO, SequenceNumber::default(), ObjectDigest::MIN),
+                    0,
+                )
+            })
+            .pipe(|data| SenderSignedData {
+                intent_message: IntentMessage::new(Intent::default(), data),
+                tx_signature: Ed25519SuiSignature::from_bytes(&[0; Ed25519SuiSignature::LENGTH])
+                    .unwrap()
+                    .into(),
+            })
+            .pipe(Self::new)
```

Comment:
The creation of the advance epoch system transaction probably looks the same, could we merge the logic?

Reply:
Yeah happy to do so